src/sentry/logging/handlers.py

```
@@ -164,3 +165,18 @@ def emit(self, record, logger=None):
         key = metrics_badchars_re.sub("", key)
         key = ".".join(key.split(".")[:3])
         metrics.incr(key, skip_internal=False)
+
+
+class SamplingFilter(logging.Filter):
+    """
+    A logging filter to sample logs with a fixed probability.
+
+    p -- probability log is emitted. Float in range [0.0, 1.0]
+    """
+
+    def __init__(self, p: float):
+        super().__init__()
+        self.sample_probability = p
```

`sample_rate` would be better I think.  And `p` should be renamed to `sample_rate`.  Also add an assert `assert 0 <= sample_rate <= 1`.

---

Do we want to raise an error in prod if user gives a bad input?

---

it wouldn't raise in prod, but much before, any time the file is imported. if we want it to be dynamic / set via options, then you can try/catch it and throw a sentry error